components/automate-cli/cmd/chef-automate/cmdUtils.go

```diff
@@ -0,0 +1,428 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/chef/automate/components/automate-deployment/pkg/cli"
+	"github.com/pkg/errors"
+)
+
+type Cmd struct {
+	PreExec   func(cmdInputs *CmdInputs, sshUtil SSHUtil, infra *AutomteHAInfraDetails, remoteService string, timestamp string, writer *cli.Writer) error
+	CmdInputs *CmdInputs
+}
+
+type CmdInputs struct {
+	Cmd         string
+	Args        []string
+	Single      bool
+	NodeIp      string
+	InputFiles  []string
+	Outputfiles []string
+	ErrorCheck  bool
+	NodeType    bool
+}
+
+type NodeTypeAndCmd struct {
+	Frontend    *Cmd
+	Automate    *Cmd
+	Chef_server *Cmd
+	Postgresql  *Cmd
+	Opensearch  *Cmd
+	Infra       *AutomteHAInfraDetails
+}
+
+type CmdResult struct {
+	HostIP      string
+	OutputFiles []string
+	Output      string
+	Error       error
+}
+
+type CmdUtil interface {
+	RunCommand() error
+	CmdExecOnNodes(input *CmdInputs, nodeIps []string, remoteService string, timestamp string, cliWriter *cli.Writer) error
+	RemoteJobs(command string, inputFiles map[string]string, outputFiles []string, remoteService string, errorCheck bool, newSSHUtil SSHUtil, resultChan chan CmdResult)
+}
+
+type CmdUtilImpl struct {
```

Comment:
Then will then become:
```suggestion
type remoteCmdExecutor struct {
```
I think if this implementation is never directly called from outside package, then we can keep it private.